src/libraries/Native/Unix/System.Native/pal_random.c

```diff
@@ -76,6 +79,17 @@ int32_t SystemNative_GetCryptographicallySecureRandomBytes(uint8_t* buffer, int3
         else
             return 0;
     }
+#elif defined(__APPLE__) && __APPLE__
+    CCRNGStatus status = CCRandomGenerateBytes(buffer, bufferLength);
```

Comment:
Would it be also beneficial to replace the `arc4random_buf` path above with this?

Reply:
Maybe? As you pointed out, `arc4random_buf` is backed by CoreCrypto on 10.15+ (using `ccrng_generate`). So I don't think we really gain much by changing it. The question then is, what do we do in `GetNonCryptographicallySecureRandomBytes` if `CCRandomGenerateBytes` fails (since it returns a status), whereas `arc4random_buf` doesn't seem to have a failure path.

I'd be hesitant to switch the `CCRandomGenerateBytes` without doing something with the return value. Fall back to `arc4random_buf`?

Reply:
`GetNonCryptographicallySecureRandomBytes` has fallback path using lrand48.

I think we can simply delete the `HAVE_ARC4RANDOM_BUF` ifdef.

Reply:
I assume that calling `CCRandomGenerateBytes` is going to be a bit than calling arc4random wrapper. It would be useful to confirm it.

Reply:
I wasn't able to observe any discernible performance differences. At least on macOS 11. Both of them call CoreCrypto's `ccrng_generate`. So:

* (CommonCrypto) `CCRandomGenerateBytes` -> (CoreCrypto) `ccrng_generate`
* (LibC) `arc4random_buf` -> (CoreCrypto) `ccrng_generate`

So they are both wrappers around Apple's (not documented) CoreCrypto. I'm happy to change it still if you would like, but I do not see an immediate benefit.

> we can simply delete the HAVE_ARC4RANDOM_BUF ifdef.

I've seen some PRs to get .NET working / building on other BSD-like operating systems. Supposedly we would want to continue to use `arc4random_buf` on those platforms?

Reply:
Fair point.